chatlearn/models/vllm_module_v2.py

```diff
@@ -321,3 +341,93 @@ def pipeline_parallel_rank(self):
         :meta private:
         """
         return get_pipeline_model_parallel_rank()
+
+    def model_setup_for_workers(self):
+        self.llm.llm_engine.model_executor._run_workers("model_setup")
+
+    # pylint: disable=unused-argument
+    def offload_for_workers(self,
+                            to_onload_weights=None,
+                            to_build_grad_buffers=None,
+                            to_onload_main_weights=None,
+                            to_onload_optimizer_states=None):
+        """
+        call offload for all workers
+        """
+        self.llm.llm_engine.model_executor._run_workers("offload")
+
+    def onload_for_workers(self,
+                           to_onload_weights=None,
+                           to_build_grad_buffers=None,
+                           to_onload_main_weights=None,
+                           to_onload_optimizer_states=None):
+        """
+        call onload for all workers
+        """
+        self.llm.llm_engine.model_executor._run_workers("onload")
+
+    def empty_cache_for_workers(self):
+        """
+        call empty cache for all workers
+        """
+        self.llm.llm_engine.model_executor._run_workers("empty_cache")
+
+    def empty_cuda_graph_for_workers(self):
+        """
+        call empty cuda_graph for all workers
+        """
+        self.llm.llm_engine.model_executor._run_workers("empty_cuda_graph")
+
+    def offload_weights(self):
+        """
+        offload weights
+        """
+        if self.module_args.offload_weights:
+            self._memory_manager.offload_weights()
+
+    def onload_weights(self):
+        """
+        onload weights
+        """
+        if self.module_args.offload_weights:
+            self._memory_manager.onload_weights()
+
+    def empty_cache(self):
+        if self.worker.gpu_cache is not None:
+            for ele in self.worker.gpu_cache: # pylint: disable=unused-variable
+                ele = None
+            self.worker.gpu_cache = None # pylint: disable=access-member-before-definition
+
+        if hasattr(self.worker, "cache_engine") and self.worker.cache_engine is not None:
+            for c_e in self.worker.cache_engine:
+                c_e.cpu_cache = None
+                c_e.gpu_cache = None
+            self.worker.cache_engine = None
+
+        self.clear_cache()
+
+    def clear_cache(self):
+        if not self.timers("gc").started_:
+            self.timers("gc").start()
+        gc.collect()
+        self.timers("gc").stop()
+
+        super().empty_cache()
```

Comment:
What's the difference between `empty_cache` and `clear_cache`? Their names are quite similar to each other.

Reply:
empty_cache use for clearing kv-cache, clear_cache is using for gc.  Refer to the function definitions in the vll_module class for naming.